payroll-processor-functions/src/payroll-processor-functions/Features/Payrolls/PayrollTrigger.cs

```diff
@@ -0,0 +1,156 @@
+using Microsoft.AspNetCore.Http;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.Azure.WebJobs;
+using Microsoft.Azure.WebJobs.Extensions.Http;
+using Microsoft.Extensions.Logging;
+using Microsoft.WindowsAzure.Storage.Queue;
+using Microsoft.WindowsAzure.Storage.Table;
+using PayrollProcessor.Functions.Features.Employees;
+using PayrollProcessor.Functions.Infrastructure;
+using System;
+using System.Threading.Tasks;
+
+namespace PayrollProcessor.Functions.Features.Payrolls
+{
+    public class PayrollTrigger
+    {
+        [FunctionName("Payroll_GetAll")]
+        public async Task<IActionResult> GetPayrolls(
+            [HttpTrigger(AuthorizationLevel.Anonymous, "GET", Route = "payrolls")] HttpRequest req,
+            [Table("payrolls")] CloudTable payrollesTable,
+            ILogger log)
+        {
+            log.LogInformation($"Retrieving all payrolls: [{req}]");
+
+            var payrollQuerier = new TableQuerier(payrollesTable);
+
+            var payrolls = await payrollQuerier.GetAllData<Payroll, PayrollEntity>(e => PayrollEntity.Map.To(e));
+
+            return new OkObjectResult(Response.Generate(payrolls));
+        }
+
+        [FunctionName("EmployeePayrolls_GetAll")]
+        public async Task<IActionResult> GetAllPayrollsForEmployee(
+            [HttpTrigger(AuthorizationLevel.Anonymous, "GET", Route = "employees/{employeeId}/payrolls")] HttpRequest req,
+            [Table("employeePayrolls")] CloudTable employeePayrollsTable,
```

Comment:
Tables/queues can be accessed by `new`'ing up a table/queue service from the SDK, but the auto-injection that Functions gives us is super convenient.

We miss out on having an easily testable abstraction, but I was able to get around this in the current unit test without _too_ much effort.